modmesh/gui/naca.py

```diff
@@ -94,16 +94,68 @@ def calc_naca4_points(number, npoint, open_trailing_edge=False,
     return ret
 
 
-def draw_naca4(world, number, npoint, fac, off_x, off_y, **kw):
+def cal_sample_points(points):
+    # this is to determine the numbe of sample points for a cubic bezier curve
+    spacing = 0.01
+    samples = [-1] * (points.shape[0] - 1)
+    for it in range(points.shape[0] - 1):
```

Comment:
You can use the [batch hypotenuse function](https://numpy.org/doc/stable/reference/generated/numpy.hypot.html) to avoid the loop.